packages/spacecat-shared-http-utils/src/auth/handlers/ims.js

```diff
@@ -57,6 +56,34 @@ const transformProfile = (payload) => {
   return profile;
 };
 
+function getTenants(profile, organizations) {
+  const contexts = profile.projectedProductContext;
+  if (!Array.isArray(contexts) || contexts.length === 0) {
+    return [];
+  }
+
+  const filteredContexts = contexts
+    .filter((context) => context.prodCtx.serviceCode === SERVICE_CODE)
+    // remove duplicates
+    .filter((context, index, array) => array.findIndex(
+      (tenant) => (context.prodCtx.owningEntity
+        && tenant.prodCtx.owningEntity === context.prodCtx.owningEntity),
+    ) === index)
+    // remove the auth source from the id (<id>@<auth-src>)
+    .map((context) => ({
+      id: context.prodCtx.owningEntity.split('@')[0],
+      subServices: context.prodCtx.enable_sub_service?.split(',').filter((subService) => subService.startsWith(SERVICE_CODE)),
+    }));
+
+  return organizations.filter(
+    (org) => filteredContexts.findIndex((ctx) => ctx.id === org.orgRef.ident) !== -1,
+  ).map((org) => ({
+    id: org.orgRef.ident,
+    name: org.orgName,
+    subServices: filteredContexts.find((ctx) => ctx.id === org.orgRef.ident)?.subServices,
+  }));
+}
```

Comment:
This has a very high complexity, due to some `findIndex` inside `filter` methods. I would reduce complexity: WDYT?
```suggestion
  const ctxById = new Map<string, string[]>();

  for (const { prodCtx } of contexts) {
    if (prodCtx.serviceCode !== SERVICE_CODE || !prodCtx.owningEntity) continue;

    const id = prodCtx.owningEntity.split('@')[0];

    if (!ctxById.has(id)) {
      const subServices = prodCtx.enable_sub_service
        ?.split(',')
        .filter((s) => s.startsWith(SERVICE_CODE));

      ctxById.set(id, subServices ?? []);
    }
  }

  return organizations
    .filter((org) => ctxById.has(org.orgRef.ident))
    .map((org) => ({
      id: org.orgRef.ident,
      name: org.orgName,
      subServices: ctxById.get(org.orgRef.ident)!,
    }));
```